test/components/embedders/types/test_protocol.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any, Dict
+
+from haystack.components.embedders.types.protocol import TextEmbedder
+
+
+class MockTextEmbedder:
```

Comment:
Let's also add one or two InvalidTextEmbedder examples that don't have `text` as an input parameter, or don't have a key "embedding" in the output, or don't have a List[float] as value of the "embedding" key in the output.

Reply:
> or don't have a key "embedding" in the output, or don't have a List[float]

I suspect this would be for the case where we would the output class `EmbeddingOutput ', but since we don't these tests don't have any impact.

Reply:
```python
class InvalidTextEmbedder:
    
    def run(self, something_else: float) -> Dict[str, Any]:
        """"
        Invalid implementation that does not return the expected keys.
        """
        return {"embedding": List[float]}  # 'str' is not a valid embedding


def test_invalid_protocol_implementation():
    embedder: TextEmbedder = InvalidTextEmbedder()
```


I'm trying to have a test like this, but surprisingly, it doesn't trigger any error